crates/core/transaction/src/is_action.rs

```diff
@@ -349,13 +361,20 @@ impl IsAction for SwapClaim {
         // Get the advice notes for each output from the swap claim
         let output_1 = txp.advice_notes.get(&self.body.output_1_commitment);
         let output_2 = txp.advice_notes.get(&self.body.output_2_commitment);
+        let nullifier = self.body.nullifier;
+        let swap_tx: Option<TransactionId> =
+            match txp.transaction_ids_by_nullifier.get(&nullifier.to_string()) {
```

Comment:
Does this work? I'd be surprised if `nullifier.to_string()` returned a Base64 string. There's in general no guarantee that the `ToString` impl on the Rust domain type has any relation to the protojson encoding. For instance `Address`'s `ToString` does bech32.